demo_fastapi/sales.py

```diff
@@ -57,27 +73,118 @@ def create_order(
         )
 
     # if everything ok, make an order
-    order_id = uuid1().int
+    order_id = random.randrange(10**8)  # 12345  # uuid1().int
     order = Order(
         id=order_id, datestamp=datestamp, buyer=buyer, apples=apples, oranges=oranges
     )
+
+    # save to the database
+    cursor = cnxn.cursor()
+    try:
+        cursor.execute(
+            "INSERT INTO ShoppingList VALUES (?,?,?,?,?)",
+            datestamp,
+            buyer,
+            apples,
+            oranges,
+            order_id,
+        )
+        cursor.commit()
+    except pyodbc.DatabaseError as err:
+        raise HTTPException(
+            status_code=500,
+            detail="Error updating database! Recheck your entries.\n" + str(err),
+        )
+
     return order
 
 
-# @app.put("/orders/{order_id}")
-# def update_order(
-#     order_id: int,
-#     datestamp: str,
-#     buyer: str,
-#     apples: PositiveInt = None,
-#     oranges: PositiveInt = None,
-# ):
-#     # update order with ID "order_id"
-#     order = Order(datestamp=datestamp, buyer=buyer, apples=apples, oranges=oranges)
-#     return {"order_id": order_id, "order": order}
+@app.put("/orders/")
+def update_order(
+    order_id: int,
+    datestamp: str = None,
+    buyer: str = None,
+    apples: PositiveInt = None,
+    oranges: PositiveInt = None,
+):
+    """update order with ID = order_id"""
+    cnxn = app.state.connection_manager.connection
+    cursor = cnxn.cursor()
+    cursor.execute("SELECT * FROM ShoppingList WHERE id = ?", order_id)
+    row = cursor.fetchone()
 
+    datestamp = datestamp if datestamp is not None else row.datestamp
+    buyer = buyer if buyer is not None else row.buyer
+    apples = apples if apples is not None else row.apples
+    oranges = oranges if oranges is not None else row.oranges
 
-# @app.get("/orders/{order_id}")
-# def read_order(order_id: int):
-#     # get data of order with ID "order_id"
-#     return {"item_id": order_id, "order": order_id}
+    try:
+        cursor.execute(
+            "UPDATE ShoppingList SET datestamp = ?, buyer = ?, apples = ?, oranges = "
```

Comment:
While this is absolutely beautiful for this small example and the practice purposes, in a production setup this can be a bit problematic. We can talk about it. Could you think of a reason?

Reply:
Let's talk about this! I wasn't sure how to do this on a bigger scale.